src/lib/updating-element.ts

```diff
@@ -462,7 +472,25 @@ export abstract class UpdatingElement extends HTMLElement {
    * @returns {Element|DocumentFragment} Returns a node into which to render.
    */
   protected createRenderRoot(): Element|ShadowRoot {
-    return this.attachShadow({mode : 'open'});
+    const shadowRoot = this.attachShadow({mode : 'open'});
+    const styles = (this.constructor as typeof UpdatingElement).styles;
+    if (window.ShadyCSS !== undefined && !(window.ShadyCSS as any).nativeShadow) {
+      // TODO(sorvell): fix type
+      (window.ShadyCSS as any).prepareAdoptedCssText(styles, this.localName);
+    } else if (supportsAdoptedStyleSheets) {
+      // TODO(sorvell): fix type
+      (shadowRoot as any).adoptedStyleSheets = styles;
+    } else {
+      this.requestUpdate();
+      this._updatePromise.then(() => {
+        styles.forEach((s) => {
+          const style = document.createElement('style');
+          style.textContent = s;
+          shadowRoot.appendChild(style);
```

Comment:
to match Constructible StyleSheet ordering these have to come first in the ShadowRoot. Is that happening because you're directly waiting `_updatePromise`? Comment if so.

Reply:
Yup, comment added. Note, they come last in the `shadowRoot`.

Reply:
Don't they have to come first in case there are `<style>` tags in the rest of the shadow content?

Reply:
Ok, I see: https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets Let's include a link to the spec